### DISCUSSION TITLE
Please add features

### DISCUSSION THREAD

**Discussion by KalpanaChetan** [ErrorFlynn/ytdlp-interface#21] [Ideas]
1.please add scheduler

Playlist downloader

3.Download video from a playlist in this method
Create a folder (channel name) and create another folder (playlist name) inside it and a third folder (video number) inside it
Video has to be downloaded in third folder
Create a third folder (video number) for each video

**Comment by ErrorFlynn** [comment]
Can I be honest with you? Adding a scheduler sounds like a lot of work :)

As for point 3 on your list, just add this as a custom argument:
`-o "%(channel)s\%(playlist_title)s\%(playlist_index)d\%(title)s.%(ext)s"`

**Reply by KalpanaChetan** [reply]
Window 's task scheduler is a great option for point 1 

The problem is that the video does not download automatically after the app is started.


Have to click on start download option every time

**Reply by ErrorFlynn** [reply]
If I understand you correctly, you're talking about adding some items to the queue, and then exiting the program (which causes the unprocessed queue items to be remembered). Then, you want the queue to start automatically the next time the program is run by Task Scheduler. That could be done very easily, by making the program accept some command line argument (like `--start-queue` or something). Is that correct?

Edit: I could even add a command for adding URLs to the queue.

**Reply by KalpanaChetan** [reply]
Yes you are right 

But it is not working ( - - start queue )

I am very simple user .

How many command line argument are used in program

**Reply by ErrorFlynn** [reply]
`--start-queue` doesn't exist yet, I was just suggesting that I could implement it in the next version. Sorry for the confusion!

Anyway, I understand what the problem is, and I'll implement a solution in the next version (whether it's a command line argument, or something else). Thanks for the feedback.

**Reply by KalpanaChetan** [reply]
Ok

**Reply by ErrorFlynn** [reply]
I've released the new version, update to it and let me know if you find any issues.

Instead of adding a command line argument, I've chosen to add a new setting: `When the program starts, automatically start processing the queue`. It's off by default so you have to go to settings and turn it on.

**Reply by KalpanaChetan** [reply]
Thank you very much 

It is working

**Reply by KalpanaChetan** [reply]
Please add playlist downloader

**Reply by ErrorFlynn** [reply]
> Please add playlist downloader

You can already download playlists, just paste the playlist URL. For example, if you paste `https://www.youtube.com/playlist?list=PLu7cY2CPiRjVY-VaUZ69bXHZr5QslKbzo`, it will add that playlist to the queue, and then you can download it (it takes a while to get info about it, but you can start the download right away). And if you use `-o "%(channel)s\%(playlist_title)s\%(playlist_index)d\%(title)s.%(ext)s"` as a custom argument, it will put the videos in the folder structure you asked about in your original post. Let me know if there are any problems.

**Reply by KalpanaChetan** [reply]
Thank you 

Now I understand why the playlist was not downloading

 https://user-images.githubusercontent.com/112402936/202851256-4e4e8adc-7b6d-4aa0-a435-073d0390ab80.mov

**Reply by KalpanaChetan** [reply]
https://www.youtube.com/playlist?list=PLu7cY2CPiRjVY-VaUZ69bXHZr5QslKbzo


Download playlist

**Reply by KalpanaChetan** [reply]
> https://www.youtube.com/playlist?list=PLu7cY2CPiRjVY-VaUZ69bXHZr5QslKbzo
> 
> 
> Download playlist




 https://m.youtube.com/watch?v=xodtfM1r9FA&list=PLu7cY2CPiRjVY-VaUZ69bXHZr5QslKbzo



Download only one video

**Comment by KalpanaChetan** [comment]
If  there are 20 videos in a playlist and we want to download only 11 to 20 videos is it possible
